### PR TITLE
fix: transparency on child windows being lost

### PR DESCRIPTION
**Pull request by nornagon**
#### Description of Change
This fixes a couple of issues with transparent windows that were introduced
with #30778.

1. Sometimes, transparent windows would be white until a repaint (e.g. resize)
   happened. This seems to be an issue with the `SetPageBaseBackgroundColor`
   API, possibly will eventually be resolved with
   https://chromium-review.googlesource.com/c/chromium/src/+/2958369 or
   similar.
2. Child windows don't have their WebPreferences set immediately, so the
   background color was nullopt at the time that HandleNewRenderFrame was
   called. We fix this by updating the background color when the WebContents
   wrapper is created in `WebContents::CreateFromWebPreferences`.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Fixed an issue where some transparent windows would show with a white background.

### CODE REVIEW

**Review by VerteDinde — COMMENTED** [review]
I think the tests are segfaulting on `BrowserWindow.getBackgroundColor() returns correct value if backgroundColor is set`, not sure why that would be, taking a look  👀

File: shell/browser/api/electron_api_web_contents.cc [hunk 1]
```diff
@@ -1373,8 +1373,10 @@ void WebContents::HandleNewRenderFrame(
   // Set the background color of RenderWidgetHostView.
   auto* web_preferences = WebContentsPreferences::From(web_contents());
   if (web_preferences) {
-    web_contents()->SetPageBaseBackgroundColor(
-        web_preferences->GetBackgroundColor());
+    absl::optional<SkColor> color = web_preferences->GetBackgroundColor();
+    web_contents()->SetPageBaseBackgroundColor(color);
+    web_contents()->GetRenderWidgetHostView()->SetBackgroundColor(
+        color.value_or(SK_ColorWHITE));
```
Comment (samuelmaddock) [inline shell/browser/api/electron_api_web_contents.cc:1379]: Maybe we should create `WebContents::SetBackgroundColor(absl::optional<SkColor> color)` to reduce the chance of this being modified only in one place.

File: shell/browser/api/electron_api_browser_window.cc [hunk 2]
```diff
@@ -362,7 +362,11 @@ void BrowserWindow::Blur() {
 
 void BrowserWindow::SetBackgroundColor(const std::string& color_name) {
   BaseWindow::SetBackgroundColor(color_name);
-  web_contents()->SetPageBaseBackgroundColor(ParseHexColor(color_name));
+  SkColor color = ParseHexColor(color_name);
+  web_contents()->SetPageBaseBackgroundColor(color);
+  auto* rwhv = web_contents()->GetRenderWidgetHostView();
+  if (rwhv)
+    rwhv->SetBackgroundColor(color);
```
Comment (samuelmaddock) [inline shell/browser/api/electron_api_browser_window.cc:369]: If we add the `WebContents::SetBackgroundColor()` method, this function could become
```cpp
void BrowserWindow::SetBackgroundColor(const std::string& color_name) {
  BaseWindow::SetBackgroundColor(color_name);
  if (api_web_contents_) {
    api_web_contents_->SetBackgroundColor(ParseHexColor(color_name));
  }
}
```

**Review by samuelmaddock — COMMENTED** [review]
Functionality looks good, just had a minor suggestion.

**Review by VerteDinde — APPROVED** [review]
Looks good! 👍

**Review by zcbenz — APPROVED** [review]

**Pull request merged**